ocean_spark/connect_hook.py

```diff
@@ -0,0 +1,105 @@
+from typing import Dict, Any
+import asyncio
+from asyncio.events import AbstractEventLoop
+from pyspark.sql import SparkSession
+
+from airflow import __version__ as airflow_version
+
+from airflow.hooks.base import BaseHook
+
+from airflow import __version__
+from airflow.exceptions import AirflowException
+
+from urllib.parse import urljoin
+
+from requests import exceptions as requests_exceptions
+
+from ocean_spark.response import ApiResponse
+
+from ocean_spark_connect.inverse_websockify import Proxy
+import threading
+
+API_HOST = "wss://api.spotinst.io/ocean/spark/"
+FE_HOST = "https://console.spotinst.com/ocean/spark/"
+
+USER_AGENT_HEADER = {"user-agent": "airflow-{v}".format(v=__version__)}
+
+DEFAULT_CONN_NAME = "ocean_spark_default"
+
+
+class OceanSparkConnectHook(BaseHook):
+    conn_name_attr: str = "ocean_spark_conn_id"
+    default_conn_name: str = "ocean_spark_default"
+    conn_type: str = "ocean_spark"
+    hook_name: str = "Ocean for Apache Spark"
+
+    def __init__(
+        self,
+        ocean_spark_conn_id: str = "ocean_spark_default",
+        app_id: str = "",
+        sql: str = "select 1",
+    ):
+        super().__init__()
+        self.conn_id = ocean_spark_conn_id
+        self.conn = self.get_connection(ocean_spark_conn_id)
+        self.token = self.conn.password
+        self.cluster_id = self.conn.host
+        self.account_id = self.conn.login
+        self.app_id = app_id
+        self.sql = sql
+        self.spark = SparkSession.builder.remote("sc://localhost").getOrCreate()
+
+    def inverse_websockify(self, url: str, loop: AbstractEventLoop) -> None:
+        proxy = Proxy(url, self.token)
+        loop.run_until_complete(proxy.start())
+        loop.run_forever()
+
+    def execute(self, app_id: str, sql: str) -> None:
```

Comment:
We should probably decide on whether we use the `app_id` parameter or the `app_id` class member

Reply:
I think we should just have it read as `execute(sql)` and build the path up below with the app_id member